class.eventcalendar.plugin.php

```diff
@@ -1,265 +1,279 @@
+
+
+   /**
+    * Adds menu entry for calendar
+    *
+    * @param object $Sender Base Controller
+    */   
+   public function Base_Render_Before($Sender) {
+      if(CheckPermission('Plugins.EventCalendar.View') && $Sender->Menu) {
+         $Sender->Menu->AddLink(T('Event Calendar'), T('EventCalendar'), 'eventcalendar');
+      }
+   }
+   
+   /**
+    * Adds input fields to new discussion form
+    * Check's for CategoryID, Add and Manage category
+    * Allows creation of events in current and next year
+    * Datefield is prefilled with current date by eventcalendar.js
+    *
+    * @param object $Sender PostController
+    */ 
+   public function PostController_BeforeBodyInput_Handler($Sender) {
+      if(!CheckPermission(array('Plugins.EventCalendar.Add', 'Plugins.EventCalendar.Manage'))) {
+         return;
+      }
+
+      $Sender->AddJsFile('eventcalendar.js', 'plugins/EventCalendar');
+	  $Sender->AddCssFile('eventcalendar.css', 'plugins/EventCalendar');
+	  $Sender->AddCssFile('jquery-ui.css', 'plugins/EventCalendar');	  	
+	  $Sender->AddJsFile('jquery-ui.js', 'plugins/EventCalendar');
+      $Sender->AddDefinition('EventCalendarCategoryIDs', json_encode(C('Plugins.EventCalendar.CategoryIDs')));
+
+      // initially don't hide elements in allowed categories
+      $CategoryID = $Sender->Discussion->CategoryID;
+      if (!in_array($CategoryID, C('Plugins.EventCalendar.CategoryIDs'))) {
+         $Hidden = ' Hidden';   
+      }
+	  
+	  //Removed when switching drop down fields to a date picker
+      //$Year = date('Y');
+      //$YearRange = $Year.'-'.($Year + 1);
+
+      $HtmlOut = <<< EOT
+<div class="P EventCalendarInput{$Hidden}">
+   {$Sender->Form->Label('Event Date', 'EventCalendarDate')}  
+	//Updated when switching drop down fields to a date picker
+   {$Sender->Form->Input('EventCalendarDate', 'EventCalendarDate')}
+</div>
+EOT;
+//{$Sender->Form->Date('EventCalendarDate', array('YearRange' => $YearRange, 'fields' => array('day', 'month', 'year')))} //from above
+
+      echo $HtmlOut;
+   } // End of PostController_BeforeBodyInput_Handler
+
+   /**
+    *  Add Validation for event date and check permissions
+    *
+    * @param object $Sender DiscussionModel
+    */
+   public function DiscussionModel_BeforeSaveDiscussion_Handler($Sender) {
+      $Session = Gdn::Session();
+      $CategoryID = $Sender->EventArguments['FormPostValues']['CategoryID'];
+      // Reset event date and return if wrong category or no right to add event
+      if (!in_array($CategoryID, C('Plugins.EventCalendar.CategoryIDs')) || !$Session->CheckPermission(array('Plugins.EventCalendar.Add', 'Plugins.EventCalendar.Manage'))) {
+         $Sender->EventArguments['FormPostValues']['EventCalendarDate'] = '';
+         return;
+      }	  
+	  //formatting the date correctly since datepicker uses a different format
+	  $time = strtotime($Sender->EventArguments['FormPostValues']['EventCalendarDate']);	  
```

Comment:
You should stick to the naming conventions. So better use $Time instead of $time. Shouldn't it be better $Date anyway?